### PR TITLE
Add new API to purge note history

### PR DESCRIPTION
**Pull request by Abhilasha06**
Signed-off-by: Abhilasha Sinha <abhisinha662000@gmail.com>

### Component/Part

Backend

### Description
This PR adds a new API to purge the history of a note. 
The API deletes all the revisions of a note except the latest one.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)

#1064

### CODE REVIEW

**Comment by ErikMichelson** [comment]
Thanks for your contribution!

File: src/revisions/revisions.service.ts
Comment (DerMolly) [inline src/revisions/revisions.service.ts:37]: This function should get it's own unit test in the `revsions.service.spec.ts` file.

**Review by DerMolly — CHANGES_REQUESTED** [review]
This was very good work. Thank you very much. I'd be happy to merge this once the requested changes were made (and the code was rebased).

File: src/revisions/revisions.service.ts
Comment (DerMolly) [reply src/revisions/revisions.service.ts:36]: I'm not sure we mention this already in the CONTRIBUTING docs, so we should probably add this there @davidmehren @InnayTool

File: src/revisions/revisions.service.ts
Comment (Abhilasha06) [reply src/revisions/revisions.service.ts:37]: Done

**Review by DerMolly — CHANGES_REQUESTED** [review]
You forgot to add the requested esdoc

**Comment by Abhilasha06** [comment]
> You forgot to add the requested esdoc
Done

**Review by DerMolly — APPROVED** [review]

File: src/revisions/revisions.service.ts
Comment (InnayTool) [reply src/revisions/revisions.service.ts:43]: I think this function should be called 
```suggestion
  async purgeRevisions(note: Note): Promise<Revision[]> {
```
or
```suggestion
  async purgeHistory(note: Note): Promise<Revision[]> {
```

File: src/revisions/revisions.service.ts
Comment (davidmehren) [inline src/revisions/revisions.service.ts:53]: Unfortunately, this code is buggy. If there is only one revision, it gets deleted.
You could fix this by using `filter` instead of `splice`:
```js
// get the latest revision, we want to keep it
const latestRevison = await this.getLatestRevision(note.id);
// get all revisions except the latest
const oldRevisions = revisions.filter(
 (item) => item.id !== latestRevison.id,
);
// delete the old revisions
return await this.revisionRepository.remove(oldRevisions);
```

Please also add a unit test for the "only one revision" case.

File: src/revisions/revisions.service.spec.ts
Comment (davidmehren) [inline src/revisions/revisions.service.spec.ts:120]: The test should also check if the expected revision remains in the 'database'.
You will need to change the mocked `remove` function to actually change the `revisions` array like this:
```js
revisionRepo.remove = jest
        .fn()
        .mockImplementation((deleteList: Revision[]) => {
          revisions = revisions.filter((item) => !deleteList.includes(item));
          return Promise.resolve(deleteList);
        });
```

**Review by davidmehren — CHANGES_REQUESTED** [review]
Thanks for your contribution!

There are two things that still need work before we can merge this.  
Please also don't forget to [clean up the commit history](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md#commit-guidelines)

File: src/revisions/revisions.service.spec.ts
Comment (davidmehren) [inline src/revisions/revisions.service.spec.ts:133]: You can merge this `describe` block with the previous one.

**Review by davidmehren — CHANGES_REQUESTED** [review]
Nearly there!

You seem to have accidentally included some other changes in your commit. You should be able to solve this by updating the `develop` branch, rebasing your branch on the updated `develop`, fixing the commit message and finally force-pushing. If you have any problems or questions, you can always ask in the dev chat :smiley:

**Review by davidmehren — APPROVED** [review]
Thanks!

**Pull request merged**
